cognite/client/data_classes/_base.py

```diff
@@ -387,6 +397,14 @@ def _load(
 
 T_CogniteResourceList = TypeVar("T_CogniteResourceList", bound=CogniteResourceList)
 
+T_WriteList = TypeVar("T_WriteList", bound=CogniteResourceList)
+
+
+class WriteableCogniteResourceList(CogniteResourceList, Generic[T_CogniteResource, T_WriteList]):
+    @abstractmethod
+    def as_write(self) -> T_WriteList:
+        raise NotImplementedError()
```

Comment:
This can be simplified to
```suggestion
class WriteableCogniteResourceList(CogniteResourceList, Generic[T_WriteClass]):
    @abstractmethod
    def as_write(self) -> CogniteResourceList[T_WriteClass]:
        raise NotImplementedError()
```

Reply:
If I do this I get `mypy` complaints on all child classes on the form
```
cognite\client\data_classes\annotations.py:369: error: "WriteableCogniteResourceList" expects 1 type argument, but 2 given  [type-arg]
    class AnnotationList(WriteableCogniteResourceList[Annotation, AnnotationWriteList]):
                         ^
```